app/controllers/sign_up/email_confirmations_controller.rb

```diff
@@ -18,6 +20,15 @@ def create
 
     private
 
+    def validate_request_id
+      result = RequestIdValidator.new(params[:_request_id]).submit
+      analytics.track_event(Analytics::EMAIL_CONFIRMATION_REQUEST_ID, result.to_h)
+
+      return if result.success?
+      flash[:error] = t('errors.messages.request_id_invalid')
+      redirect_to sign_up_email_resend_url
```

Comment:
I'm not sure this is the right thing to do. We want them to double check that they clicked the link in the email, or if they copied and pasted it, that they did so correctly. By redirecting them to resend the email confirmation, we lose the request_id, and they won't be redirected back to the SP when they are done.

I think we should leave them on the password creation page with the flash message that explains what the error is.